cmake/Macros.cmake

```diff
@@ -25,17 +25,20 @@ macro(now RESULT)
   endif(WIN32)
 endmacro(now)
 
+find_package(Git)
+
 macro(git_id RESULT)
-  if(WIN32)
-    execute_process(COMMAND "cmd" " /C git --git-dir=${PROJECT_SOURCE_DIR}/.git rev-parse --short=8 HEAD" OUTPUT_VARIABLE ${RESULT})
-    string(REGEX REPLACE "(.*)\n" "\\1" ${RESULT} ${${RESULT}})
-  elseif(UNIX)
-    execute_process(COMMAND "git" "--git-dir=${PROJECT_SOURCE_DIR}/.git" "rev-parse" "--short=8" "HEAD" OUTPUT_VARIABLE ${RESULT})
+  if (GIT_FOUND)
+    if(WIN32)
+      execute_process(COMMAND "cmd" " /C ${GIT_EXECUTABLE} --git-dir=${PROJECT_SOURCE_DIR}/.git rev-parse --short=8 HEAD" OUTPUT_VARIABLE ${RESULT})
+    else()
+      execute_process(COMMAND "${GIT_EXECUTABLE}" "--git-dir=${PROJECT_SOURCE_DIR}/.git" "rev-parse" "--short=8" "HEAD" OUTPUT_VARIABLE ${RESULT})
+    endif()
     string(REGEX REPLACE "(.*)\n" "\\1" ${RESULT} ${${RESULT}})
-  else(WIN32)
-    message(SEND_ERROR "Git ID implemented")
+  else()
+    message(WARNING "Git executable not found, revision number not available.")
     set(${RESULT} 0)
-  endif(WIN32)
+  endif()
```

Comment:
Ah yes, good point, I'm sure git will throw some error but not clear what happens then.  We could check for `.git` folder in source root?  I'm not sure that's 100% reliable but should help.  I'll also see what happens when it just errors.